--- a/node_modules/react-joyride/dist/index.mjs
+++ b/node_modules/react-joyride/dist/index.mjs
@@ -1,7 +1,7 @@
 import e, { memo as t, forwardRef as n, useState as r, useReducer as o, useMemo as i, useEffect as s, useLayoutEffect as c, useCallback as l } from "react";
-import a, { findDOMNode as u, unmountComponentAtNode as p, unstable_renderSubtreeIntoContainer as d } from "react-dom";
+import { findDOMNode as u } from "react-dom";
+import { createRoot as d } from "react-dom/client";
 import { S as m, T as f, C as g, a as h, b as y, P as v, R as b, I as w, d as E, O as S, E as C, _ as x } from "./_commonjsHelpers-B86b14d8.js";
 import k from "is-mobile";
 import { S as T } from "./scroll-B1n6gH4w.js";
@@ -482,14 +482,16 @@
       this.tooltip = document.createElement("div"), this.tooltip.className = "joyride-portal", document.body.appendChild(this.tooltip), this.renderTooltip();
     }
     componentDidUpdate() {
-      this.renderTooltip();
+      this.renderTooltip();  
     }
     componentWillUnmount() {
-      p(this.tooltip), document.body.contains(this.tooltip) && document.body.removeChild(this.tooltip);
+      this.root.unmount(), document.body.contains(this.tooltip) && document.body.removeChild(this.tooltip);
     }
     renderTooltip() {
       const { children: e, ...t } = this.props, n = e;
-      d(n, this.tooltip);
+      this.root ? this.root.render(n) : this.root = d(this.tooltip, {
+        onRecoverableError: () => {}
+      }), this.root.render(n)
     }
     render() {
       return null;
