--- a/node_modules/react-joyride/dist/index.mjs
+++ b/node_modules/react-joyride/dist/index.mjs
@@ -1,7 +1,6 @@
-import ReactDOM from 'react-dom';
 import React, { useReducer, useEffect, useMemo, useCallback, useRef, useState } from 'react';
-import { unmountComponentAtNode, unstable_renderSubtreeIntoContainer } from 'react-dom';
+import { createRoot } from 'react-dom/client';
 import PropTypes from 'prop-types';
 import { Animate } from 'react-simple-animate';
 import Scroll from 'react-scroll';
@@ -213,20 +212,11 @@
   }, [props.run]);
   var renderPortal = useCallback(function (portal) {
     if (!isMounted.current) return;
-    unstable_renderSubtreeIntoContainer(instance, portal, portalTarget.current, function () {
-      var hasListeners = addListener(this);
-
-      if (!hasListeners) {
-        addListener(this);
-      }
-    });
+    createRoot(portalTarget.current).render(portal);
   }, [instance]);
   var unmountPortal = useCallback(function () {
     if (!portalTarget.current) {
       return;
     }
-
-    unmountComponentAtNode(portalTarget.current);
+    createRoot(portalTarget.current).unmount();
   }, []);
   useEffect(function () {
     if (!portalTarget.current) {
